Fixed is_window_minimized() and _maximized()

NikolaBerovic opened the pull request:
Since old version of this function returns if window is minimized/maximized through code, this one returns if it was minimized/maximized by windows title bar icons.

Old code always returns false if minimized/maximized is done by title bar icons

EDIT: it returns if window is minimized/maximized in general (not only by titlebar, just for better understanding)

nikolaphoenix committed:
Fixed is_window_minimized() and _maximized()

Since old version of this function returns if window is minimized/maximized through code, this one returns if it was minimized/maximized by windows title bar icons.

Hinsbart commented on platform/windows/os_windows.cpp:
```diff
@@ -1673,8 +1673,14 @@ void OS_Windows::set_window_minimized(bool p_enabled){
+	wp.length = sizeof(wp);
+	GetWindowPlacement(hWnd, &wp);
 
+	if (wp.showCmd == SW_MINIMIZE)
```
Maybe this could be shortened to `return wp.showCmd == SW_MINIMIZE`?

NikolaBerovic replied:
for sure, works just as fine
updated

27thLiz reviewed (approved):
Looks good :)

nikolaphoenix committed:
just cleaner return

nikolaphoenix committed:
Update os_windows.cpp

nikolaphoenix committed:
updated return condition

updated from conditions:
SW_MAXIMIZE
SW_MINIMIZE
to:
SW_SHOWMAXIMIZED
SW_SHOWMINIMIZED

maximized worked fine, but minimized didn't.

Hinsbart commented:
Great, but that's a lot of commits now for a small change.
@NikolaBerovic  could you squash or rebase them together, by any chance? :)

NikolaBerovic commented:
yeah i wanted to do that, but i'm quite new to github, how do i do that?

akien-mga commented:
The changes look mostly fine but there's a problem, as the function can now report something different from the value contained by the `minimized` variable, and for example this function would still not behave as one would expect: https://github.com/godotengine/godot/blob/8325a8c77c2943187d2bfcc5ddfd9c6284ebf50b/platform/windows/os_windows.cpp#L225-L228

Instead of checking the actual Windows API when querying the state, there should be something listening to a callback from Windows when the window is minimized or maximized, and the cached boolean should be updated accordingly, so that it's always correct.

akien-mga commented:
I don't want to merge this one for the reasons exposed above, it would make the API inconsistent to what it is on other platforms: return whether Godot thinks of the window as minimized or maximized. Those methods are just getters for a private boolean, what should be fixed if anything is catching the minimized/maximized event sent by the system and properly update the internal booleans.
